supervisor/shared/usb/usb_desc.c

```diff
@@ -68,8 +69,10 @@ static supervisor_allocation *device_descriptor_allocation;
 static supervisor_allocation *configuration_descriptor_allocation;
 static supervisor_allocation *string_descriptors_allocation;
 
-static const char manufacturer_name[] = USB_MANUFACTURER;
-static const char product_name[] = USB_PRODUCT;
+static char manufacturer_name[127] = USB_MANUFACTURER;
+static char product_name[127] = USB_PRODUCT;
```

Comment:
Hm I see... That does make things a bit trickier, but I think I found the snippet you mean. I'll need to expand `usb_get_boot_py_data()`, `usb_return_boot_py_data()` and `usb_boot_py_data_size()`.

Reply:
Actually, we're lucky and we don't need to store the strings after boot.py finishes.

The `static` variables I declared above are only used to store the strings until they are converted into UTF-16 and stored in the descriptor area alongside the other existing values. This conversion happens in `usb_build_descriptors()` which is called in `usb_return_boot_py_data()` just after the boot.py VM has been shut down.

So in `boot.py` we can dynamically allocate the space for the descriptor strings (in ASCII). Since `boot.py` runs very briefly and doesn't do very much, it should be enough to allocate the maximum (2*128 bytes, including trailing zeros) once. When the VM shuts down, the strings are briefly copied into the stack by `main.c` but upon returning through `usb_return_boot_py_data` they are just passed on to `usb_build_descriptors` and then thrown away.

Reply:
Sorry, I somehow missed your replies here. Yes, the storage managment you describe sounds like a fine plan.